src/components/Modal.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useRef, useEffect } from 'react';
+import ReactDom from 'react-dom';
+import Button from './Button';
+import styles from './Modal.module.scss';
+const modalRoot = document.getElementById('portal');
+
+const ModalDefault = ({ children, modalIsOpen, setModalIsOpen, ...props }) => {
+  const modalRef = useRef(null);
+  const closeModalButtonRef = useRef(null);
+
+  useEffect(() => {
+    if (modalIsOpen) {
+      closeModalButtonRef.current.focus();
+
+      modalRef.current.addEventListener('transitionend', () => {
+        closeModalButtonRef.current.focus();
+      });
+    }
+  }, [modalIsOpen]);
```

Comment:
While I greatly admire and appreciate this solution, it doesn't work very well - switching focused elements with `Tab` key doesn't put focus on elements in correct order. This causes a serious accessibility issue.

Since focus trapping is pretty difficult thing to do, I think it would be okay to use an external library for that purpose. Please see [focus-trap-react](https://github.com/focus-trap/focus-trap-react) and try to implement it with your modal component.